packages/device/src/components/register.component.tsx

```diff
@@ -1,11 +1,65 @@
+import { createSignal, onMount } from 'solid-js';
+import * as QRCode from 'qrcode';
 import { Device } from '../classes';
-import { PincodeComponent } from './pincode.component';
+import styles from './register.module.css';
+import logoSvg from '../assets/logo.svg';
 
 export function RegisterComponent(props: { device: Device; pincode: string }) {
+  const [qrCodeUrl, setQrCodeUrl] = createSignal<string>('');
+
+  onMount(async () => {
+    try {
+      // Generate QR code for the registration URL
+      // In a real implementation, this would be the actual registration URL
+      const registrationUrl = `https://app.castmill.com/register/${props.pincode}`;
+      const qrDataUrl = await QRCode.toDataURL(registrationUrl, {
+        width: 200,
+        margin: 1,
+        color: {
+          dark: '#1e3a8a',
+          light: '#ffffff'
+        }
+      });
+      setQrCodeUrl(qrDataUrl);
+    } catch (error) {
+      console.error('Error generating QR code:', error);
+    }
+  });
+
   return (
-    <>
-      <h1>Register</h1>
-      <PincodeComponent pincode={props.pincode} />
-    </>
+    <div class={styles.container}>
+      <div class={styles.main}>
+        <header class={styles.header}>
+          <img src={logoSvg} alt="Castmill" class={styles.logo} />
+          <h1 class={styles.title}>Register Your Device</h1>
+          <p class={styles.subtitle}>Connect your device to the Castmill platform</p>
+        </header>
+
+        <div class={styles.content}>
+          <div class={styles.pincodeSection}>
```

Comment:
As I mentioned, I think it was better to have a separate component for the pincode part as we had before.